website_sale_checkout_adj/controllers/main.py

```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+
+class RequireZipCode(WebsiteSale):
+
+    def _get_mandatory_billing_fields(self):
+      res = super(RequireZipCode, self)._get_mandatory_billing_fields()
+      return res + ["zip"]
+
+    def _get_mandatory_shipping_fields(self):
+      res = super(RequireZipCode, self)._get_mandatory_shipping_fields()
+      return res + ["zip"]
```

Comment:
These changes are not truly generic in that it forces users to key in a post code regardless of the country (for example, there is no postal codes in Hong Kong). The real generic solution would be to judge whether it's required or not according to the country, in a similar manner as how it's done for `state_id` (see https://github.com/odoo/odoo/blob/28c3f51c4878fbcd79b2e819948465fcf2160ebc/addons/website_sale/controllers/main.py#L477-L480).

In this case, however, I conclude that the current logic should be OK since:
- I think there shouldn't be many customers that do not have a postal code (for those with no postal code, they can just key in '0000' or something and it won't be a big issue).
- The cost of implementing the "real" solution can be a bit high for the benefits (we need to override the entire `checkout_form_validate()` method).

Reply:
I see.
In this pull-request, adopt current change.

I'll add solution in future.